aiida/orm/importexport.py

```diff
@@ -1888,12 +1909,28 @@ def import_data_sqla(in_path, user_group=None, ignore_unknown_nodes=False,
             if pks_for_group:
                 # If user specified a group, import all things in it
                 if user_group:
-                    group = user_group[0]
+                    if isinstance(user_group, str):
+                        # Will try to retrieve Group with provided label
+                        res_group = Group.objects.find(label=user_group)
+                        if res_group:
+                            # Use existing Group
+                            group = res_group[0]
+                        else:
+                            # Create new import Group with provided label
+                            group = Group(label=user_group, type_string=IMPORTGROUP_TYPE)
+                            session.add(group.backend_entity._dbmodel)
+                    elif isinstance(user_group, Group):
+                        # Use provided Group
+                        group = user_group
+
+                        # Make sure provided Group is stored
+                        if not group.is_stored:
```

Comment:
Shouldn't it be stored here then? (Note I think that all of this logic should be removed anyway and just accept an existing group in the arguments)

Reply:
It should - Will remove this as well